### PR TITLE
OD-490 [Feature] Data and Time fields preferences improvements

### PR DESCRIPTION
**Pull request by AndrRyaz**
@sofiiakvasnevska 

## Issue
OD-490 https://weboo.atlassian.net/browse/OD-490
## Description
Added new fields to radio and remove unnecessary fields
## Screenshots/screencasts
http://somup.com/crhIoDFMMs
## Backward compatibility
This change is fully backward compatible.
## Reviewers

### CODE REVIEW

File: js/components/date.js [hunk 1]
```diff
@@ -72,12 +72,17 @@ Fliplet.FormBuilder.field('date', {
...
+        }
 
         $vm.updateValue();
       });
 
-      this.datePicker.datepicker('setDate', new Date(this.value) || new Date());
+      if (this.autofill !== 'empty') {
+        this.datePicker.datepicker('setDate', new Date(this.value) || new Date());
+      }
+
```
Comment (tonytlwu) [inline js/components/date.js:85]: No empty lines at the end of `{}`

File: js/components/date.js [hunk 2]
```diff
@@ -90,12 +95,19 @@ Fliplet.FormBuilder.field('date', {
       this.empty = false;
     }
 
+    if(this.autofill === 'empty') {
```
Comment (tonytlwu) [inline js/components/date.js:98]:
```suggestion
    if (this.autofill === 'empty') {
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Rejected + 1

The requirement specifies:

> When the user chooses **one of the default value types to autofill with current date/time**, the When applying the current date/time field should appear.

The default value types that autofill with the current date/time are:

* Autofill with current date if it's empty
* Always autofill with current date

In 0:49 of http://somup.com/crhIoDFMMs you clicked on **Autofill with current date if it's empty**, but the **When applying the current date** field did not appear. This needs to be corrected.

File: js/components/date.js (hunk 2 above)
Comment (AndrRyaz) [reply js/components/date.js]: fixed

File: js/components/date.js (hunk 1 above)
Comment (AndrRyaz) [reply js/components/date.js]: fixed

**Comment by AndrRyaz** [comment]
@tonytlwu new video - http://somup.com/crhZiTqjO3

**Review by tonytlwu — CHANGES_REQUESTED** [review]
@AndrRyaz In 0:10 of https://screencast-o-matic.com/watch/crhZiTVfS7n it shows that you clicked on **Autofill with a specific date if it's empty**. According to https://weboo.atlassian.net/browse/ID-904 Fig. 2 and specification 1b. users should be able shown a date picker and then be able to select a date. But in the video, no additional field was shown.

Rejected +1

**Pull request closed without merging**